### PR TITLE
TS Migration

### PR DESCRIPTION
**Pull request by Hyperkid123**
requires: https://github.com/RedHatInsights/insights-chrome/pull/1797

### CODE REVIEW

File: src/js/sentry.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import * as Sentry from '@sentry/browser';
+import { ChromeUser } from './types';
```
Comment (karelhala) [inline src/js/sentry.ts:2]: Could be better to indicate we're importing `type`. Based on the doc, `import type` will be completely removed once transpiled.
```suggestion
import type { ChromeUser } from './types';
```

File: src/js/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+export type VisibilityFunctions = {
+  isOrgAdmin: () => Promise<boolean>;
+  isActive: () => Promise<boolean>;
+  isInternal: () => Promise<boolean>;
+  isEntitled: () => Promise<boolean>;
+  isProd: () => boolean;
+  isBeta: () => boolean;
+  isHidden: () => true;
+  withemai: (toHave: string) => Promise<boolean>;
```
Comment (karelhala) [inline src/js/types.d.ts:55]: This should be `withEmail` and `toHave` should be unknon amount of attrs.
```suggestion
  withEmail: (...toHave: string[]) => Promise<boolean>;
```

File: src/js/utils.ts [hunk 3]
```diff
@@ -112,9 +113,9 @@ export function lastActive(searchString, fallback) {
   }, fallback);
 }
 
-export const isAnsible = (sections) => sections.includes('ansible') && sections.includes('insights');
+export const isAnsible = (sections: string[]) => (sections.includes('ansible') && sections.includes('insights') ? 1 : 0);
```
Comment (karelhala) [inline src/js/utils.ts:116]: Any reason to switch from `boolean` to `1`|`0`?

File: src/js/sentry.ts (hunk 1 above)
Comment (Hyperkid123) [reply src/js/sentry.ts:2]: We now import it form the types package

File: src/js/types.d.ts (hunk 2 above)
Comment (Hyperkid123) [reply src/js/types.d.ts]: Type is now imported from the types package.

**Review by karelhala — APPROVED** [review]

**Pull request merged**